core/src/main/java/org/axonframework/commandhandling/disruptor/DisruptorCommandBus.java

```diff
@@ -137,78 +150,119 @@
     private final CommandTargetResolver commandTargetResolver;
     private final int publisherCount;
     private final MessageMonitor<? super CommandMessage<?>> messageMonitor;
+    private final Disruptor<CommandHandlingEntry> disruptor;
+    private final CommandHandlerInvoker[] commandHandlerInvokers;
+
     private volatile boolean started = true;
     private volatile boolean disruptorShutDown = false;
 
     /**
-     * Initialize the DisruptorCommandBus with given resources, using default configuration settings. Uses a Blocking
-     * WaitStrategy on a RingBuffer of size 4096. The (2) Threads required for command execution are created
-     * immediately. Additional threads are used to invoke response callbacks and to initialize a recovery process in
-     * the case of errors.
-     */
-    public DisruptorCommandBus() {
-        this(new DisruptorConfiguration());
-    }
-
-    /**
-     * Initialize the DisruptorCommandBus with given resources and settings. The Threads required for command
-     * execution are immediately requested from the Configuration's Executor, if any. Otherwise, they are created.
+     * Instantiate a {@link DisruptorCommandBus} based on the fields contained in the {@link Builder}. The Threads
+     * required for command execution are immediately requested from the Configuration's Executor, if any. Otherwise,
+     * they are created.
+     * <p>
+     * Will assert that the {@link CommandTargetResolver}, {@link MessageMonitor}, {@link RollbackConfiguration},
+     * {@link ProducerType}, {@link WaitStrategy} and {@link Cache} are not {@code null}. Additional verification is
+     * done on the the {@code coolingDownPeriod}, {@code publisherThreadCount}, {@code bufferSize} and
+     * {@code invokerThreadCount} to check whether they are positive numbers. If any of these checks fails, an
+     * {@link AxonConfigurationException} will be thrown.
      *
-     * @param configuration The configuration for the command bus
+     * @param builder the {@link Builder} used to instantiate a {@link DisruptorCommandBus} instance
      */
     @SuppressWarnings("unchecked")
-    public DisruptorCommandBus(DisruptorConfiguration configuration) {
-        Assert.notNull(configuration, () -> "configuration may not be null");
-        Executor executor = configuration.getExecutor();
+    protected DisruptorCommandBus(Builder builder) {
+        builder.validate();
+
+        dispatchInterceptors = new CopyOnWriteArrayList<>(builder.dispatchInterceptors);
+        invokerInterceptors = new CopyOnWriteArrayList<>(builder.invokerInterceptors);
+        publisherInterceptors = new ArrayList<>(builder.publisherInterceptors);
```

Comment:
Should not be a `CopyOnWriteArrayList`?

Reply:
Dunno actually, just kept it as is for this PR to be honest. @abuijze / @m1l4n54v1c, what do you think? Old typo?